checkers/todoCommentWithoutDetail_checker.go

```diff
@@ -0,0 +1,52 @@
+package checkers
+
+import (
+	"go/ast"
+	"regexp"
+	"strings"
+
+	"github.com/go-critic/go-critic/checkers/internal/astwalk"
+	"github.com/go-critic/go-critic/framework/linter"
+)
+
+func init() {
+	var info linter.CheckerInfo
+	info.Name = "todoCommentWithoutDetail"
+	info.Tags = []string{"diagnostic", "experimental"}
+	info.Summary = "Detects TODO comments without detail/assignee"
+	info.Before = `
+// TODO
+fiiWithCtx(nil, a, b)
+`
+	info.After = `
+// TODO(admin): pass context.TODO() instead of nil
+fiiWithCtx(nil, a, b)
+`
+	collection.AddChecker(&info, func(ctx *linter.CheckerContext) (linter.FileWalker, error) {
+		visitor := &todoCommentWithoutCodeChecker{
+			ctx:   ctx,
+			regex: regexp.MustCompile(`^\s*(TODO|FIX|FIXME|BUG)\s*$`),
+		}
+		return astwalk.WalkerForComment(visitor), nil
+	})
+}
+
+type todoCommentWithoutCodeChecker struct {
+	astwalk.WalkHandler
+	ctx   *linter.CheckerContext
+	regex *regexp.Regexp
+}
+
+func (c *todoCommentWithoutCodeChecker) VisitComment(cg *ast.CommentGroup) {
+	comment := strings.TrimSpace(cg.Text())
```

Comment:
Using `Text()` method is generally slower than using `cg.List[...].Text` fields directly.
I think we can analyze it in line-by-line fashion and avoid TrimSpace call as well since regexp already includes a space skipping part.
What do you think?

Reply:
@quasilyte Thanks, I've addressed this. Sorry for the delay. It looks like using the `cg.List[...].Text` fields include the comment initialisers (`//` and `/*`) so I've adjusted the regex to take this into account and everything seems to work.